packages/common/src/hardhat/tasks/test.js

```diff
@@ -18,14 +15,12 @@ internalTask(TASK_TEST_GET_TEST_FILES, async (_, { config, network }, runSuper)
   // Build absolute path for all directories on user-specified whitelist.
   const whitelist = config.networks[network.name].testWhitelist;
   if (whitelist && Array.isArray(whitelist)) {
-    const whitelistDirs = whitelist.map((x) => path.resolve(TESTS_DIR, x));
-
     filePaths = filePaths.filter((filePath) => {
-      for (let whitelistedDir of whitelistDirs) {
-        if (!path.resolve(process.cwd(), filePath).startsWith(whitelistedDir)) return false;
+      for (let whitelistString of whitelist) {
+        if (filePath.includes(whitelistString)) return true;
```

Comment:
So this changes the whitelist behavior such that any file that includes the whitelisted string is included? Previously it had to start with the whitelisted string IIUC.

Is my understanding correct?

Reply:
correct. Are you ok with that change? it just makes defining the whitelist a bit simpler to reason about and simplified the code.

Reply:
I'm good with this!